common/test.c

```diff
@@ -501,7 +503,12 @@ p11_test_copy_setgid (const char *input)
 		return NULL;
 	}
 
-	path = strdup ("/tmp/test-setgid.XXXXXX");
+	tmpdir = getenv("TMPDIR");
+	if (tmpdir == NULL)
+		tmpdir = "/tmp";
+
+	snprintf(file, sizeof(file), "%s/test-setgid.XXXXXX", tmpdir);
+	path = strdup (file);
```

Comment:
The construction of ```path``` could be simplified using ```asprintf```, which we use extensively and provide a fallback definition for portability.